### PR TITLE
Fetch AWS Secrets using batch method [ENG-23893]

### PR DESCRIPTION
**Pull request by plenza**
Refactor to use boto3 get batch AWS SecretManager values.

### CODE REVIEW

File: safe_init/secrets.py [hunk 1]
```diff
@@ -45,31 +93,109 @@ def resolve_secrets() -> Mapping[str, str | None]:
...
+    Returns:
+        A tuple containing:
+        - A dictionary of secrets retrieved from the cache.
+        - A list of ARNs for secrets that are not in the cache.
+    """
+    secrets_in_cache = {}
+    secrets_not_in_cache = []
 
+    for _, secret_arn in secret_arns.items():
+        secret_value = get_secret_from_cache(_strip_json_key_prefix_if_present(secret_arn))
```
Comment (pczubak) [inline safe_init/secrets.py:115]: Can't we also use batch get here? Redis has support for it.

Comment (pczubak) [reply safe_init/secrets.py:115]: Also, can't we "secrets_client" inside "get_secret_from_cache"?

File: safe_init/secrets.py [hunk 2]
```diff
@@ -35,6 +45,44 @@ def resolve_secrets() -> Mapping[str, str | None]:
...
+                error_message = f"Failed to retrieve secrets: {errors}"
+                raise SecretResolutionError(error_message, errors)  # noqa: TRY301
+
+            for secret_arn, secret_value in fetched_secrets.items():
+                save_secret_in_cache(secret_arn, secret_value)
+                secrets[secret_arn] = secret_value
+        except Exception as e:
+            if bool_env("SAFE_INIT_FAIL_ON_SECRET_RESOLUTION_ERROR"):
+                raise
+            log_warning("Failed to resolve some secrets", exc_info=e)
```
Comment (pczubak) [inline safe_init/secrets.py:67]: Should this be log_error?

**Review by pczubak — APPROVED** [review]

File: safe_init/secrets.py (hunk 2 above)
Comment (mLupine) [reply safe_init/secrets.py:67]: It's consistent with the current implementation, which is a warning. If we want it to fail, there's the env var that raises if secret resolution fails

File: safe_init/secrets.py (hunk 1 above)
Comment (mLupine) [reply safe_init/secrets.py:115]: > Also, can't we "secrets_client" inside "get_secret_from_cache"?

What's the benefit? We currently have a clear separation of functions responsible for fetching secrets — one from AWS, one from cache — and it allows the function implementing the logic to decide which one to use

**Review by mLupine — APPROVED** [review]
Nice

**Pull request merged**